src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -307,65 +321,79 @@ protected function guardAgainstInvalidTranslationResetRequest($content)
     /**
      * @param $lang
      * @param $type
-     * @param $selected
+     * @param string $theme Selected theme name
      * @param null $search
      *
      * @return array
      */
-    private function getNormalTree($lang, $type, $selected, $search = null)
+    private function getNormalTree($lang, $type, $theme, $search = null)
```

Comment:
If `getNormalTree` is supposed to only be called for Themes, I suggest we strongly validate the input by checking `$theme` actually exists and refers to an installed theme, or throw an exception.

Reply:
(maybe not in a patch version though 😛)

Reply:
Since we planned a refacto maybe it should be done in the refacto?

Reply:
This is called on line 159 with a null theme (`BO` translations and `other`)